### PR TITLE
Fix handicap overlays in order to fit to the basemap

### PR DESCRIPTION
**Pull request by gevos**
Please, have a look at the commit message of this [commit](https://github.com/geops/trafimage-maps/pull/573/commits/e40bb061e0e3ba5dded95633e789606a28fccf38)

# How to

Test branch with dev data: https://github.com/geops/trafimage-maps/pull/624
Should also work with prod data (preview app in this PR)

Then switch to handicap topic

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] IE11 tested.
- [x] The title means something for a human being.
- [ ] Labels applied. if it's a release? a hotfix?
- [ ] Tests added.

### CODE REVIEW

File: src/layers/layerHelper.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ function getDataResolution(resolution) {
 
 function getGeneralization(resolution) {
   const res = getDataResolution(resolution);
-  return { 750: 5, 500: 10, 250: 30, 100: 30, 75: 100, 50: 100 }[res] || 150;
+  return (
+    { 750: 5, 500: 10, 250: 30, 100: 30, 75: 100, 50: 100, 20: 100 }[res] || 150
```
Comment (gevos) [inline src/layers/layerHelper.js]: Core change

File: src/layers/HandicapLayer/HandicapLayer.js [hunk 2]
```diff
@@ -84,16 +84,11 @@ class HandicapLayer extends VectorLayer {
    * @returns {Object|null}
    */
   style(feature, resolution) {
-    let geometry = feature.getGeometry();
```
Comment (gevos) [inline src/layers/HandicapLayer/HandicapLayer.js]: Default geometry will not be used anymore

**Comment by edellucien** [comment]
LGTM :+1:

File: src/config/layers.js [hunk 3]
```diff
@@ -379,8 +380,8 @@ export const gemeindegrenzen = new TrafimageGeoServerWMSLayer({
       },
       tileGrid: new TileGrid({
         extent: projectionExtent,
-        resolutions,
-        matrixIds: resolutions.map((r, i) => `${i}`),
+        resolutions: LayerHelper.mapResolutions,
+        matrixIds: LayerHelper.mapResolutions.map((r, i) => `${i}`),
```
Comment (gevos) [inline src/config/layers.js:384]: Move map resolutions to LayerHelper

File: src/layers/HandicapLayer/HandicapLayer.js [hunk 4]
```diff
@@ -119,18 +119,14 @@ class HandicapLayer extends VectorLayer {
...
-      const wkt = (feature.get('generalizations') || {})[`geom_gen${gen}`];
-      geometry = wkt ? this.wktFormat.readGeometry(wkt.split(';')[1]) : null;
-    }
+    const gen = LayerHelper.getMapboxGeneralization(resolution);
+    const wkt = (feature.get('generalizations') || {})[`geom_gen${gen}`];
+    const geometry = wkt
+      ? this.wktFormat.readGeometry(wkt.split(';')[1])
+      : null;
+
+    const minVisibility = LayerHelper.getMapboxDataResolution(resolution) * 10;
```
Comment (gevos) [inline src/layers/HandicapLayer/HandicapLayer.js:128]: Use special Mapbox generalization and data resolution

File: src/config/layers.js (hunk 3 above)
Comment (friedjoff) [reply src/config/layers.js:384]: Why not use the new helper function for all layers referencing `resolutions` and delete the `resolutions` variable in this file?

File: src/layers/layerHelper.js [hunk 5]
```diff
@@ -28,4 +52,80 @@ function getOldGeneralization(resolution) {
...
+      250: 30,
+      100: 30,
+      50: 100,
+      20: 100,
+    }[res] || 150
+  );
+}
+
+export default {
+  mapResolutions,
```
Comment (friedjoff) [inline src/layers/layerHelper.js:125]:
```suggestion
  getMapResolutions,
```

Comment (gevos) [reply src/layers/layerHelper.js]: Changed

File: src/config/layers.js (hunk 3 above)
Comment (gevos) [reply src/config/layers.js]: I have replaced `mapResolutions` by `getMapResolutions()`

File: src/config/layers.js [hunk 6]
```diff
@@ -39,29 +40,6 @@ const projectionExtent = [
   20037508.3428,
 ];
 
-const resolutions = [
```
Comment (gevos) [inline src/config/layers.js:42]: Core change

File: src/layers/layerHelper.js [hunk 7]
```diff
@@ -1,3 +1,35 @@
+// Map resolutions by zoom (array index)
+const mapResolutions = [
```
Comment (gevos) [inline src/layers/layerHelper.js:2]: Core change

**Comment by gevos** [comment]
Introduce getMapboxDataResolution() and getMapboxGeneralization() for HandicapLayer
    
This fixes the bug where the handicap data did not match
with the underlying basemap (netzkarte_personenverkehr_v2).
    
This bugfix considers
- the new mapping between zoom, generalization level, and minimum visibility resp. resolutions 
  (e.g. with resolutions 4500).
- the zoom level substraction in the react-spatial MapboxLayer (why?),
  see https://github.com/geops/react-spatial/blob/939d9d2535f44b1086e10a187fbb5dc800ae3336/src/layers/MapboxLayer.js#L54

**Pull request merged**